test/any/FetchWebIDLFile-test.js

```diff
@@ -0,0 +1,113 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+describe('FetchWebIDLFile', function() {
+  var IDLFile;
+  var GithubIDLFile;
+  var GitilesIDLFile;
+  var payloadContent = 'Some sort of file content';
+
+  beforeAll(function() {
+    IDLFile = foam.lookup('org.chromium.webidl.IDLFile');
+    GithubIDLFile = foam.lookup('org.chromium.webidl.GithubIDLFile');
+    GitilesIDLFile = foam.lookup('org.chromium.webidl.GitilesIDLFile');
+  });
+
+  beforeEach(function() {
+    foam.CLASS({
+      package: 'org.chromium.webidl.test',
+      name: 'MockHTTPRequest',
+      extends: 'foam.net.HTTPRequest',
+
+      methods: [
+        function send() {
+          if (this.url == 'http://test.url/someFile.idl') {
```

Comment:
Another way to do this would be to declare and register the class in each test individually, with no `if`s at all.